e2e/testcases/helm_sync_test.go

```diff
@@ -809,8 +817,174 @@ func TestHelmARTokenAuth(t *testing.T) {
 	}
 }
 
+// TestHelmEmptyChart verifies Config Sync can apply an empty Helm chart.
+//
+// Requirements:
+// 1. Helm auth https://cloud.google.com/artifact-registry/docs/helm/authentication
+// 2. gcloud auth login OR gcloud auth activate-service-account ACCOUNT --key-file=KEY-FILE
+// 3. Artifact Registry repo: us-docker.pkg.dev/${GCP_PROJECT}/config-sync-test-private
+// 4. GKE cluster with Workload Identity
+// 5. Google Service Account: e2e-test-ar-reader@${GCP_PROJECT}.iam.gserviceaccount.com
+// 6. IAM for the GSA to read from the Artifact Registry repo
+// 7. IAM for the test runner to write to Artifact Registry repo
+// 8. gcloud & helm
+func TestHelmEmptyChart(t *testing.T) {
+	nt := nomostest.New(t,
+		nomostesting.SyncSource,
+		ntopts.Unstructured,
+		ntopts.RequireGKE(t),
+	)
+
+	// Verify workload identity is enabled on the cluster
+	expectedPool := fmt.Sprintf("%s.svc.id.goog", *e2e.GCPProject)
+	if workloadPool, err := getWorkloadPool(nt); err != nil {
+		nt.T.Fatal(err)
+	} else if workloadPool != expectedPool {
+		nt.T.Fatalf("expected workloadPool %s but got %s", expectedPool, workloadPool)
+	}
+
+	// Validate that the GCP Service Account exists.
+	// GSA is used for CS to read the Helm Chart from Artifact Registry.
+	// The local test user will also need to be able to push to Artifact Registry.
+	gsaEmail := gsaARReaderEmail()
+	if _, err := describeGCPServiceAccount(nt, gsaEmail, *e2e.GCPProject); err != nil {
+		nt.T.Fatalf("failed to get service account for workload identity: %v", err)
+	}
+
+	// Create temp dir & delete on exit
+	tempDir, err := os.MkdirTemp("", fmt.Sprintf("%s-", nt.T.Name()))
+	if err != nil {
+		nt.T.Fatalf("failed to create temp directory: %v", err)
+	}
+	nt.T.Cleanup(func() {
+		if err := os.RemoveAll(tempDir); err != nil {
+			nt.T.Errorf("failed to delete temp dir: %v", err)
+		}
+	})
```

Comment:
We don't need to create the tempDir. The local chart can be copied to a subfolder of `nt.TmpDir`, which is deleted in the end.

Reply:
Done. RemoteHelmChart uses TmpDir.